vcd/resource_vcd_lb_app_rule.go

```diff
@@ -0,0 +1,227 @@
+				},
+				Description: "The script for the application rule. Each line as a separate array element",
+			},
+		},
+	}
+}
+
+func resourceVcdLBAppRuleCreate(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	LBRule, err := getLBAppRuleType(d)
+	if err != nil {
+		return fmt.Errorf("unable to create load balancer app rule type: %s", err)
+	}
+
+	createdPool, err := edgeGateway.CreateLBAppRule(LBRule)
+	if err != nil {
+		return fmt.Errorf("error creating new load balancer app rule: %s", err)
+	}
+
+	err = setLBAppRuleData(d, createdPool)
+	if err != nil {
+		return err
+	}
+	d.SetId(createdPool.ID)
+	return nil
+}
+
+func resourceVcdLBAppRuleRead(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	readLBRule, err := edgeGateway.ReadLBAppRuleByID(d.Id())
+	if err != nil {
+		d.SetId("")
+		return fmt.Errorf("unable to find load balancer app rule with ID %s: %s", d.Id(), err)
+	}
+
+	return setLBAppRuleData(d, readLBRule)
+}
+
+func resourceVcdLBAppRuleUpdate(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	updateLBRuleConfig, err := getLBAppRuleType(d)
+	if err != nil {
+		return fmt.Errorf("could not create load balancer app rule type for update: %s", err)
+	}
+
+	updatedLBRule, err := edgeGateway.UpdateLBAppRule(updateLBRuleConfig)
+	if err != nil {
+		return fmt.Errorf("unable to update load balancer app rule with ID %s: %s", d.Id(), err)
+	}
+
+	if err := setLBAppRuleData(d, updatedLBRule); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceVcdLBAppRuleDelete(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	err = edgeGateway.DeleteLBAppRuleByID(d.Id())
+	if err != nil {
+		return fmt.Errorf("error deleting load balancer app rule: %s", err)
+	}
+
+	d.SetId("")
+	return nil
+}
+
+// resourceVcdLBAppRuleImport is responsible for importing the resource.
+// The d.Id() field as being passed from `terraform import _resource_name_ _the_id_string_ requires
+// a name based dot-formatted path to the object to lookup the object and sets the id of object.
```

Comment:
A bit confused when reading this without context for the first time. Consider rephrasing. Maybe in a more declarative form and with a numbered list (1, 2, 3 ...)? I.e. what happens after what with some sample values.

Reply:
Changed. Any better?

Reply:
A lot! Thanks!